chore(docs): add highlight post for secrets in disk buffers (#18994)
vectordotdev/vector: website/content/en/highlights/2023-10-30-secrets-in-disk-buffers.md

the same API key when sending those events back out to a compatible service. This allows users, for
example, to set up Vector as an aggregator for all of their Datadog Agent processes, reusing the
original API key, or keys, as the events are then forwarded to the Datadog API.

## Disk buffers and event secrets and metadata

Prior to [#18816](https://github.com/vectordotdev/vector/pull/18816), these event secrets (and other
event metadata) were not stored when using disk buffers. This represented a loss of functionality
when users switched from the default in-memory buffers to disk buffers. In order to bring this
functionality up to par, we added support for storing event secrets/metadata when writing events to
disk buffers.

Naturally, event secrets represent sensitive data such as API keys and more. However, Vector
currently stores these event secrets **unencrypted** in disk buffers.

## Do I need to worry about this change?

Firstly, if you're **not** using disk buffers, then there is no change to Vector's behavior and you can
stop reading here.

There are two main scenarios where a configuration might now start storing secrets in disk buffers:

> - using a source component which has the ability to store secrets

Reviewer: ```suggestion
- When you are using a source component which has the ability to store secrets
```

> - using `remap` and adding secrets directly to events

Reviewer: ```suggestion
- When you are using `remap` and adding secrets directly to events
```

### Source components that can store secrets

> Some source components will store secrets (specifically, API keys) on an event in order to

Reviewer: ```suggestion
Some source components store secrets (specifically, API keys) on an event in order to
```

facilitate Vector acting similarly to a proxy, using as much of the original request/event data as
possible. Only two sources currently provide such behavior:

- `datadog_agent` source (stores the `DD-API-KEY` header value; **enabled** by default)
- `splunk_hec` source (stores the `Authorization` header value; **disabled** by default)

However, for both of these sinks, this behavior can be disabled by setting `store_api_key` to
`false` for the `datadog_agent` source, or setting `store_hec_token` to `false` for the `splunk_hec`
source.

> ### Manually-stored secrets via `remap`

Reviewer: ```suggestion
### Manually-stored secrets using `remap`
```

When using the `remap` transform, VRL exposes helper functions to set secrets on events. If your

> `remap` usage includes setting secrets, then those secrets would also now in scope for getting

Reviewer: ```suggestion
`remap` usage includes setting secrets, then those secrets would also now be in scope for getting
```

stored in disk buffers.

## Securing disk buffers

As mentioned above, secrets will now be stored in disk buffer data files, and will be
**unencrypted**. The data directory that Vector is configured to use should be locked down as

> tightly as possible so that only the user/group that the Vector process is run as has read/write

Reviewer: ```suggestion
tightly as possible so that only the user/group that runs the Vector process has read/write
```
